.circleci/config.yml

```diff
@@ -1,43 +1,117 @@
-# Python CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-python/ for more details
-#
-version: 2.0
+# Python CircleCI 2.1 configuration file.
+version: 2.1
+orbs:
+  gcp-gke: circleci/gcp-gke@1.0.4
+  go: circleci/go@1.3.0
+  codecov: codecov/codecov@1.1.0
 
 references:
 
-  install_deps: &install_deps
+  make_docs: &make_docs
     run:
-      name: Install Dependences
+      name: Make Documentation
       command: |
-        sudo apt-get update && sudo apt-get install -y cmake
-        pip install -r requirements/base.txt -q
-        pip install -r requirements/test.txt -q
+        # First run the same pipeline as Read-The-Docs
+        # apt-get update && apt-get install -y cmake
+        # using: https://hub.docker.com/r/readthedocs/build
+        # we need to use py3.7 ot higher becase of an issue with metaclass inheritence
+        pyenv global 3.7.3
+        python --version
+        pip install -r requirements/docs.txt
+        cd docs; make clean; make html --debug --jobs 2 SPHINXOPTS="-W"
 
-  tests: &tests
+  checkout_ml_testing: &checkout_ml_testing
    run:
-     name: Testing
+     name: Checkout ml-testing-accelerators
      command: |
-       python --version ; pip --version ; pip list
-       python -m pytest pytorch_lightning -v --junitxml=test-reports/pytest_junit.xml  --ignore=pytorch_lightning/loggers/comet.py --ignore=pytorch_lightning/loggers/mlflow.py --ignore=pytorch_lightning/loggers/neptune.py --ignore=pytorch_lightning/loggers/test_tube.py --ignore=pytorch_lightning/loggers/wandb.py --ignore=pytorch_lightning/metrics/sklearns.py
-     no_output_timeout: 15m
+       git clone https://github.com/GoogleCloudPlatform/ml-testing-accelerators.git
+       cd ml-testing-accelerators
+       git fetch origin 5e88ac24f631c27045e62f0e8d5dfcf34e425e25:stable
+       git checkout stable
+       cd ..
 
-  make_docs: &make_docs
+  build_push_docker: &build_push_docker
+    run:
+      name: Build and push Docker image
+      command: |
+        gcloud --quiet auth configure-docker
+        cd dockers/tpu-tests
+        # TODO: How to find the GITHUB_REF in CircleCI?
+        # $CI_PULL_REQUEST seems to be of form: https://github.com/org/repo-name/pull/11.
+        # Grab the last bit, e.g. pull/11, convert to pull/11/head, and use it
+        # for the GITHUB_REF so Docker can pull the latest pending code in PR.
+        git_ref=$(echo "$CI_PULL_REQUEST" | sed "s/.*pytorch-lightning\///")/head
+        docker build --tag "$GCR_IMAGE_PATH:$CIRCLE_WORKFLOW_JOB_ID" -f Dockerfile --build-arg "TEST_IMAGE=1" --build-arg "GITHUB_REF=$git_ref" .
+        #docker build --tag "$IMAGE:$GITHUB_RUN_ID" -f Dockerfile --build-arg "GITHUB_REF=$GITHUB_REF" --build-arg "TEST_IMAGE=1" .
+        docker push "$GCR_IMAGE_PATH:$CIRCLE_WORKFLOW_JOB_ID"
```

Comment:
just in case shall we add name `circleci` here to not interfere with similar images pushed from GH action?

Reply:
I think that makes sense. However I think it's pretty safe as-is since `$CIRCLE_WORKFLOW_JOB_ID` will probably never be the same as [`$GITHUB_RUN_ID`](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/workflows/tpu-testing.yml#L64). The odds are especially low since there is only a problem if CircleCI and Github Actions run at the same time and produce the same ID

I think it also makes sense to remove or disable the Github Action since the CircleCI has more coverage. We can re-enable the Github Action and remove the CircleCI job once Github Action adds support to share env vars with forked PRs

Reply:
have tried to edit it then it was hanging, not finished, mind check it?

Reply:
I have dropped GH action for PRs and let's keep it only for master commits so we keep both updated :]

Reply:
It's not necessary to add the `circleci-`

Github Actions I use `$GITHUB_RUN_ID` as the tag. These IDs look like this:
`148065161`

CircleCI I use `$CIRCLE_WORKFLOW_JOB_ID` as the tag. These IDs look like this:
`bd7593e6-b299-4f45-b74c-920539ff28e0`